### PR TITLE
synergy with draw instanced seqs

### PR DESCRIPTION
**Pull request by sinkingsugar**
This PR is a partial synergy with GFX instanced drawing.
Where we pass a sequence of matrices/locations in one block activation.

### CODE REVIEW

**Comment by sinkingsugar** [comment]
@Kryptos-FR pinged you for review mostly to give a chance to check out the codebase and the changes I did, not expecting you to understand all the bits so don't need to go to deep 😄

**Comment by Kryptos-FR** [comment]
@sinkingsugar You know well that my reply is going to be "where are the documentation and comments?"

**Comment by sinkingsugar** [comment]
> @sinkingsugar You know well that my reply is going to be "where are the documentation and comments?"

It was a huge fight with the borrow checker, one thing I noticed is I write much less comments in rust due to refactoring every iteration. Maybe I'll get better at rust one day... 
But good point, let me add some comments, at least in those places that were tedious.

**Review by Kryptos-FR — COMMENTED** [review]
I see a lot of rust files are several 1000s of LoC. Would it be possible to split them into smaller files? I find it hard to reason about such large files.

File: rust/src/blocks/physics/rigidbody.rs [hunk 1]
```diff
@@ -10,19 +10,23 @@ use crate::blocks::physics::SHAPES_VAR_TYPE;
 use crate::blocks::physics::SHAPE_TYPE;
 use crate::blocks::physics::SHAPE_VAR_TYPE;
 use crate::blocks::physics::SIMULATION_TYPE;
+use crate::core::deriveType;
```
[the remark this answers is not in the document]
Comment (sinkingsugar) [reply rust/src/blocks/physics/rigidbody.rs:13]: Yeah, I like grouping indeed but the linter/helper automatically just adds singles :/ wish it was smarter.
But there must be a tool to clean up those things I guess...

**Comment by sinkingsugar** [comment]
> I see a lot of rust files are several 1000s of LoC. Would it be possible to split them into smaller files? I find it hard to reason about such large files.

Yeah sure, I kinda like big files actually but I agree that with a project scaling up probably having more granularity in the code files is for the best.
Requires time though.

**Pull request merged**